### PR TITLE
Next Previous Button block - Add layout support

### PR DESCRIPTION
**Pull request by gigitux**
## What

This PR enables the layout property. In this way, it is possibile change the vertical alignment of the next and previous buttons. 
## Why

In this way, block themes can change the button's position via `theme.json`.

The feature is full working, we should spend some time on the following:

- Understand why all the controls are visible even if the they should not be visible according with the block.json configuration.
- ~~Ensure that it works with WordPress 6.2.~~ https://github.com/woocommerce/woocommerce-blocks/pull/10668#issuecomment-1695576831
- Add E2E tests.

## Testing Instructions

_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Open the Site Editor and edit the Single Product Template.
2. Add the Product Gallery.
3. Focus on the Next And Previous Button block.
4. Change the vertical alignment.
5. Save.
6. Visit a product page.
7. Ensure that the vertical alignment matches.
8. Install WordPress 6.2.
9. Open the Site Editor and edit the Single Product Template.
10. Focus on the Next And Previous Button block.
11. Ensure that the Vertical Alignment settings aren't visible.

* [ ] Do not include in the Testing Notes 
* [ ] Should be tested by the development team exclusively 

## Screenshots or screencast

https://github.com/woocommerce/woocommerce-blocks/assets/4463174/8b1bd4db-07a0-472d-a8f0-fb41081d8d5d

## WooCommerce Visibility

Required:

* [ ] WooCommerce Core
* [ ] Feature plugin
* [x] Experimental
* [ ] N/A

## Checklist

Required:
* [x] This PR has either a `[type]` label or a `[skip-changelog]` label.
* [ ] This PR is assigned to a milestone.

Conditional:
* [ ] This PR has a changelog description (if `[skip-changelog]` label is not present).
* [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR adds/removes an experimental interfaces, and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR has been accessibility tested.
* [ ] This PR has had any necessary documentation added/updated.

## Changelog

> N/A

### CODE REVIEW

File: assets/js/blocks/product-gallery/inner-blocks/product-gallery-large-image-next-previous/block.json [hunk 1]
```diff
@@ -9,5 +9,16 @@
     "keywords": [ "WooCommerce" ],
 	"usesContext": [ "nextPreviousButtonsPosition", "productGalleryClientId" ],
     "textdomain": "woo-gutenberg-products-block",
+	"supports": {
+		"layout": {
```
Comment (gigitux) [inline assets/js/blocks/product-gallery/inner-blocks/product-gallery-large-image-next-previous/block.json:13]: For adding the support to WordPress 6.2, I think that it is necessary do this change:
```suggestion
		"__experimentalLayout": {
```

**Comment by danieldudzic** [comment]
I have opened two issues relating to this PR in the Gutenberg repo:

1. Addressing the visible, unnecessary `Justification` settings:
https://github.com/WordPress/gutenberg/issues/53871

2. The mismatch between the current buttons default position, vs the default Gutenberg `Vertical Alignment` setting:
https://github.com/WordPress/gutenberg/issues/53872

File: assets/js/blocks/product-gallery/inner-blocks/product-gallery-large-image-next-previous/block.json (hunk 1 above)
Comment (danieldudzic) [reply assets/js/blocks/product-gallery/inner-blocks/product-gallery-large-image-next-previous/block.json:13]: Do you think we need to support WordPress 6.2. with this?

Seems like WooCommerce Blocks currently requires at least WP 6.3.

Update: Slack discussion: p1692780683893349-slack-C02FL3X7KR6

Comment (danieldudzic) [reply assets/js/blocks/product-gallery/inner-blocks/product-gallery-large-image-next-previous/block.json:13]: I have tested this with WP `6.2.2` and `__experimentalLayout` and noticed that it's not working correctly out of the box:
Ks0GBAQrVu

**Comment by gigitux** [comment]
Thanks @danieldudzic for continuing to work on this PR! 

Thinking about it, I believe we can avoid adding the experimental flag to support WordPress 6.2. The users that want to customize the position of the arrows can:
- Install Gutenberg
- Update to WordPress 6.3 or later.

In this way, we can reduce the maintenance of cost of this feature.

**Review by danieldudzic — APPROVED** [review]
Code looks solid and the functionality also works correctly.

I'd only add to the testing steps testing additionally with WP 6.2.2 and that the lack of `Vertical Alignment` settings with that version is expected.

(I use the WP Downgrade plugin to test with older versions of WP and it works great).
